src/Microsoft.Azure.SignalR.Management/ClientInvocation/WeakCallerClientResultsManager.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#if NET7_0_OR_GREATER
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Linq;
+using Microsoft.Azure.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR;
+
+namespace Microsoft.Azure.SignalR.Management
+{
+    internal sealed class WeakCallerClientResultsManager : ICallerClientResultsManager, IInvocationBinder
+    {
+        private readonly ConcurrentDictionary<string, PendingInvocation> _pendingInvocations = new();
+        private readonly string _clientResultManagerId = Guid.NewGuid().ToString("N");
+        private long _lastInvocationId = 0;
+        private readonly AckHandler _ackHandler;
+        private readonly IServiceEndpointManager _endpointManager;
+        private readonly IEndpointRouter _endpointRouter;
+        private readonly IHubProtocolResolver _hubProtocolResolver;
+
+
+        public WeakCallerClientResultsManager(IServiceEndpointManager serviceEndpointManager, IEndpointRouter endpointRouter, AckHandler ackHandler, IHubProtocolResolver hubProtocolResolver)
+        {
+            _endpointManager = serviceEndpointManager;
+            _endpointRouter = endpointRouter;
+            _ackHandler = ackHandler;
+            _hubProtocolResolver = hubProtocolResolver;
+        }
+
+        public string GenerateInvocationId(string connectionId)
+        {
+            return $"{connectionId}-{_clientResultManagerId}-{Interlocked.Increment(ref _lastInvocationId)}";
+        }
+
+        public Task<T> AddInvocation<T>(string hub, string connectionId, string invocationId, CancellationToken cancellationToken)
+        {
+            var tcs = new TaskCompletionSourceWithCancellation<T>(
+                cancellationToken,
+                () => TryCompleteResult(connectionId, CompletionMessage.WithError(invocationId, "Canceled")));
+
+            var serviceEndpoints = _endpointManager.GetEndpoints(hub);
+            var expectedAckCount = _endpointRouter.GetEndpointsForConnection(connectionId, serviceEndpoints).Count();
+
+            var multiAck = _ackHandler.CreateMultiAck(out var ackId);
+
+            _ackHandler.SetExpectedCount(ackId, expectedAckCount);
```

Comment:
this is not correct behavior, since `_endpointRouter.GetEndpointsForConnection` value might change from time to time. We need to fix this logic first.  @JialinXin